### ISSUE TITLE
indentation_linter: multiline expression with braced block

### ISSUE THREAD

**Issue by klmr** [r-lib/lintr#1944]
Consider the following code containing a multi-line expression with a braced block:

```r
condition_a() &&
  condition_b() && {
    vec <- complex_prep_work()
    any(map_lgl(vec, some_predicate))
  }
```

‘lintr’ says:

> ```
> multiline-expression.r:3:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
>     vec <- complex_prep_work()
>   ~^
> ```

I’m honestly not sure how this *should* be formatted in general; especially since intuitively I would decrease the indentation (and make ‘lintr’ happy) when `condition_b()` does not exist; i.e. I would write:

```r
condition_a() && {
  vec <- complex_prep_work()
  any(map_lgl(vec, some_predicate))
}
```

The inconsistency can be avoided by putting the `{` on its own line; unfortunately this leads to a (legitimate) complaint by the `brace_linter`.

**Comment by AshesITR** [comment]
FWIW, `styler` agrees with `lintr` on this one:
``` r
styler::style_text("
condition_a() &&
  condition_b() && {
    vec <- complex_prep_work()
    any(map_lgl(vec, some_predicate))
  }
")
#> 
#> condition_a() &&
#>   condition_b() && {
#>   vec <- complex_prep_work()
#>   any(map_lgl(vec, some_predicate))
#> }
```

<sup>Created on 2023-04-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I agree more with your style, although I'm not a huge fan of the brace at all in this case.
I'd probably write

```r
condition_a() &&
  condition_b() && 
  local({
    vec <- complex_prep_work()
    any(map_lgl(vec, some_predicate))
  })
```

Which avoids bleeding `vec` into the parent scope (?)

**Comment by MichaelChirico** [comment]
personally I think that's a really good sign you need a helper function

**Comment by klmr** [comment]
@MichaelChirico Ah, I previously had a comment explaining why I didn’t do that, but failed to move that when I made a new issue.

In a nutshell, the reason I didn’t make this into a function here is that it’s entirely single-use, and I had trouble finding good name for the function. Granted, this is a pretty niche situation. I wouldn’t usually write code like this.

**Comment by MichaelChirico** [comment]
perfect case for `# nolint` then :)
